sdk/python/feast/infra/online_stores/contrib/hazelcast_online_store/hazelcast_online_store.py

```diff
@@ -0,0 +1,230 @@
+                self._client = HazelcastClient(
+                    cluster_name=config.cluster_name,
+                    statistics_enabled=True,
+                    ssl_enabled=True,
+                    ssl_cafile=config.ca_path,
+                    ssl_certfile=config.cert_path,
+                    ssl_keyfile=config.key_path,
+                    ssl_password=config.ssl_password
+                )
+            else:
+                self._client = HazelcastClient(
+                    statistics_enabled=True,
+                    cluster_members=config.cluster_members,
+                    cluster_name=config.cluster_name
+                )
+        return self._client
+
+    @log_exceptions_and_usage(online_store="hazelcast")
+    def online_write_batch(
+            self,
+            config: RepoConfig,
+            table: FeatureView,
+            data: List[
+                Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+            ],
+            progress: Optional[Callable[[int], Any]]
+    ) -> None:
+
+        online_store_config = config.online_store
+        assert isinstance(online_store_config, HazelcastOnlineStoreConfig)
+
+        client = self._get_client(online_store_config)
+        project = config.project
+
+        query = f"""
+            SINK INTO {_feature_map_name(project, table)}
+            (entity_key, feature_name, feature_value, event_ts, created_ts)
+            values (?, ?, ?, ?, ?);
+        """
+        for entity_key, features, timestamp, created_ts in data:
+            entity_key_bin = serialize_entity_key(
+                entity_key,
+                entity_key_serialization_version=2,
+            ).hex()
+            timestamp = _convert_to_naive_utc(timestamp)
+            if created_ts is not None:
+                created_ts = _convert_to_naive_utc(created_ts)
+            for feature_name, feature_value in features.items():
+                client.sql.execute(query,
+                                   entity_key_bin,
+                                   feature_name,
+                                   feature_value.SerializeToString().hex(),
+                                   timestamp,
+                                   created_ts)
+            if progress:
+                progress(1)
+
+    @log_exceptions_and_usage(online_store="hazelcast")
+    def online_read(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        entity_keys: List[EntityKeyProto],
+        requested_features: Optional[List[str]] = None
+    ) -> List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]]:
+
+        online_store_config = config.online_store
+        assert isinstance(online_store_config, HazelcastOnlineStoreConfig)
+
+        client = self._get_client(online_store_config)
+
+        result: List[Tuple[Optional[datetime], Optional[Dict[str, Any]]]] = []
+        requested_features_str = ", ".join(f"'{f}'" for f in requested_features)
```

Comment:
We have to add this `requested_features` check either in here or in line 160 since user might want to read only some of the features, not all of them. Notice that if we check this at line 160 (after retrieving the features), there is a cost of retrieving unnecessary feature.

Reply:
The format conversion between `requested_features` to `requested_features_str` is below:

```
requested_features = ["acc_percentage", "conv_percentage"]
requested_features_str = "('acc_percentage', 'conv_percentage')"
```
This makes possible to use `IN` statement in SQL query.